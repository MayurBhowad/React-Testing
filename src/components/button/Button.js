import React, { Component } from 'react';
import './button.css';

function Button({ lable }) {

    return <div data-testid="button" className="button-style" >Hi {lable}</div>
}

export default Button
