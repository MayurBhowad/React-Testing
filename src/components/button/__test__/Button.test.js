import React from 'react';
import ReactDOM from 'react-dom';
import { isTSAnyKeyword } from '@babel/types';
import Button from './../Button';

import { render, cleanup } from '@testing-library/react';
// import 'jest-dom/extend-expect';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup)

it("Render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div)
})

it("render button correctly", () => {
    const { getByTestId } = render(<Button lable="Click me please"></Button>)
    expect(getByTestId('button')).toHaveTextContent("Click me please")
})

it("render button correctly", () => {
    const { getByTestId } = render(<Button lable="Save"></Button>)
    expect(getByTestId('button')).toHaveTextContent("Save")
})

it('match snapshot 1', () => {
    const tree = renderer.create(<Button lable="save"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('match snapshot 2', () => {
    const tree = renderer.create(<Button lable="Click me Please"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
})