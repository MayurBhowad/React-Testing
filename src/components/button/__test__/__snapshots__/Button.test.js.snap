// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  save
</div>
`;

exports[`match snapshot 2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  Hi 
  Click me Please
</div>
`;
